src/main/java/net/apnic/whowas/history/ObjectIndex.java

```diff
@@ -1,10 +1,20 @@
 package net.apnic.whowas.history;
 
 import java.util.Optional;
+import java.util.stream.Stream;
 
 /**
  * Look up object histories by ObjectKey.
  */
-public interface ObjectIndex {
+public interface ObjectIndex
+{
+    /**
+     * Provides the history for a single ObjectKey.
+     *
+     * @param objectKey ObjectKey to recieve History for.
+     * @return
+     */
     Optional<ObjectHistory> historyForObject(ObjectKey objectKey);
+
+    Stream<ObjectHistory> historyForObject(Stream<ObjectKey> objectKeys);
```

Comment:
Should this be a stream of optional ObjectHistory??

(If not, them I'm also concerned that it's a different enough overload to not warrant being an overload)

Reply:
The reason its not because the valid return for that case is an empty stream. Where the other case your option is return null for nothing found.

Reply:
When is the empty stream a valid return? If any (or all) ObjectKey from the input stream is missing ObjectHistory what should this be doing?

(I admit to assuming it was doing a straight mapping - in my defence the only implementation is indeed doing a map)

Reply:
Yeah there is a chance after looking at the code on to which a null element could be imbedded in the stream. This is not ideal. I will fix so that only non null objects will be in the stream or the stream will be empty.

Reply:
Done & Done. Have added a filter which stops null values being returned in the stream.